ricardocosta/bank-ui
# Use API to render navigation

## Summary

This PR adds dynamic navigation to the `shell` application.

We use the `/internal/navigation` API to get the items for the sidebar. Those items are also used to populate the routes provided to `RouterProvider`'s configuration.





## Testing


**Setup**
- Run the mock server locally with: `yarn nx run shell:jsonServer`
- In another terminal tab, run the application with: `yarn nx run shell:dev`
- Open the browser on: `http://localhost:3002`

**Validation**
- Notice that the application is redirected to `/dashboard`
- Modify `apps/shell/server/navigation.json` file to change an entry's `name`
- Run `yarn nx run shell:seedDb` to update the "database"
- Re-run the server with `yarn nx run shell:jsonServer`
- Refresh the application and make sure the new name is visible in the sidebar

## Other Information

**Breaking Changes [YES / NO]**  No
**Affected Component**  `shell`
